maven-compat/src/main/java/org/apache/maven/artifact/repository/metadata/DefaultRepositoryMetadataManager.java

```diff
@@ -267,19 +266,17 @@ private boolean loadMetadata(
      * TODO share with DefaultPluginMappingManager.
      */
     protected Metadata readMetadata(File mappingFile) throws RepositoryMetadataReadException {
-        Metadata result;
 
-        try (Reader reader = ReaderFactory.newXmlReader(mappingFile)) {
-            MetadataXpp3Reader mappingReader = new MetadataXpp3Reader();
-
-            result = mappingReader.read(reader, false);
+        MetadataXpp3Reader mappingReader = new MetadataXpp3Reader();
+        try (InputStream in = Files.newInputStream(mappingFile.toPath())) {
+            Metadata result = mappingReader.read(in, false);
+            return result;
         } catch (FileNotFoundException e) {
```

Comment:
I don't see the benefit of the change.
`mappingReader.read( Files.newInputStream( mappingFile.toPath() ), false )` is exactly the same as `mappingReader.read( ReaderFactory.newXmlReader( mappingFile ), false )`.
The reason is that the `read` method accepting an `InputStream` does create an xml reader using `ReaderFactory.newXmlReader`.

Reply:
That is indeed something else that needs to be changed. There is no reason to convert an input stream to a reader immediately before passing it to an XML parser, and in fact that introduces an additional way things can break if the encoding is guessed wrong. 

Until that can be done too, this at least removes a couple more dependencies on plexus-util.